packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -896,7 +896,7 @@ export class DeltaManager
         const message: IDocumentMessage = {
             clientSequenceNumber: ++this.clientSequenceNumber,
             contents: JSON.stringify(contents),
-            metadata,
+            metadata: JSON.stringify(metadata),
```

Comment:
I have a feeling this will break existing SPO / attachment blob flow.
If that's the case, a chance needs to be done first on SPO and propagate through rings to support this. Only then we can make change on client.
Can you please reach out to Marcelo to clarify this? This is the only place where I believe we have dependency on service.

That all said, doesn't it defeat the purpose? I.e. if service needs to parse it anyway, is there difference in string vs. json?

Reply:
What might be useful here is to consider client vs server metadata, as I believe we use it for different purposes - metadata for batches (I believe this is client only data) and metadata for service (blob attachment info).

Reply:
Thanks! I will follow up with Marcelo first.

The problem for us is not parsing, but storing. We prefer to always store primitive types in the DB cause storing JSONs created issues for us before. Also FRS is thinking about a long term cheaper storage for ops and the one we are considering (Azure table storage) only supports primitive types for fields.

Another option for us is to translate on the server/driver. But that will make us more inefficient as we will have to do this for every op. Also I don't want core protocol to change between two services.

Reply:
> What might be useful here is to consider client vs server metadata, as I believe we use it for different purposes - metadata for batches (I believe this is client only data) and metadata for service (blob attachment info).

Yes that would be perfect! We should come up with a separate field for anything that client wants server to parse.